Dockerfile

```diff
@@ -23,16 +24,18 @@ RUN apt-get update \
 
 # Composer setup starts here. The zip extension is required for that.
 # See https://github.com/composer/docker/blob/8a2a40c3376bac96f8e3db2f129062173bff7734/1.6/Dockerfile
+# and https://github.com/composer/docker/blob/942dac53831e7b4ca4419a135304e177b9d29dbd/1.8/Dockerfile
+# or more recently https://getcomposer.org/download/
 RUN docker-php-ext-install zip
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
 ENV COMPOSER_HOME /tmp
-ENV COMPOSER_VERSION 1.6.2
+ENV COMPOSER_VERSION 1.8.0
 
-RUN curl -s -f -L -o /tmp/installer.php https://raw.githubusercontent.com/composer/getcomposer.org/b107d959a5924af895807021fcef4ffec5a76aa9/web/installer \
+RUN curl --silent --fail --location --retry 3 --output /tmp/installer.php --url https://raw.githubusercontent.com/composer/getcomposer.org/d3e09029468023aa4e9dcd165e9b6f43df0a9999/web/installer \
     && php -r " \
-    \$signature = '544e09ee996cdf60ece3804abc52599c22b1f40f4323403c44d44fdfdd586475ca9813a858088ffbc1f233e9b180f061'; \
-    \$hash = hash('SHA384', file_get_contents('/tmp/installer.php')); \
+    \$signature = '93b54496392c062774670ac18b134c3b3a95e5a5e5c8f1a9f115f203b75bf9a129d5daa8ba6a13e2cc8a1da0806388a8'; \
+    \$hash = hash_file('sha384', '/tmp/installer.php'); \
```

Comment:
https://github.com/composer/docker/blob/942dac53831e7b4ca4419a135304e177b9d29dbd/1.8/Dockerfile references commit [b107d959a5924af895807021fcef4ffec5a76aa9](https://github.com/composer/getcomposer.org/commit/b107d959a5924af895807021fcef4ffec5a76aa9) and is composer version 1.5.2.
[The official download page](https://getcomposer.org/download/) links the current release (1.8.0), has the new hash and also switched to `hash_file`. This is done in a similar way on their ["How do I install Composer programmatically?" page](https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md).
Since I did want to go for a specific (known-working) version I linked GitHub instead of their most-recent link which also has the benefit that we can ship a known-good signature.

The signature crosscheck is [here](https://composer.github.io/pubkeys.html).

Reply:
Thanks! This provides me more context.